pages/product/[code].js

```diff
@@ -6,12 +6,84 @@ import { atomWithStorage, createJSONStorage } from "jotai/utils";
 import { useAtom } from "jotai";
 import allergens from "../../allergens.json";
 import additives from "../../additives.json";
+import { SVGIcon } from "@/components/SVGIcon";
 
 const StyledImage = styled(Image)`
+  margin: 2rem;
   width: 25%;
   height: 25%;
   object-fit: cover;
 `;
+
+const StyledProductName = styled.h2`
+  padding-top: 1rem;
+  font-weight: bold;
+  font-size: 1.2rem;
+  text-align: center;
+`;
+
+const StyledCheck = styled.div`
+  font-weight: bold;
+  display: flex;
+  flex-direction: column;
+  gap: 2rem;
+  height: 100px;
+  width: 200px;
+  padding-top: 20px;
+  padding-right: 2rem;
+  margin-top: 2rem;
+`;
+const StyledProductCard = styled.section`
+  background-color: white;
+  margin-left: 2rem;
+  margin-right: 2rem;
+  border-radius: 0.7rem;
+  box-shadow: 0px 0px 10px 2px rgba(128, 128, 128, 0.25);
+  height: 50%;
+`;
+const StyledBackButton = styled.div`
+  position: fixed;
+  left: 0.1rem;
+  top: 0.3rem;
+  color: white;
+`;
+
+const NoProduct = styled.p`
+  font-size: 35px;
+  left: 5rem;
+  top: 200px;
+  position: absolute;
+  z-index: 4;
+`;
+const StyledAllCards = styled.div`
+  margin-top: 4rem;
+`;
+const StyledIngredientsCard = styled.div`
+  margin-top: 0.5rem;
+  background-color: white;
+  margin-left: 4rem;
+  margin-right: 4rem;
+  border-radius: 0.4rem;
+  box-shadow: 0px 0px 10px 2px rgba(128, 128, 128, 0.25);
+  height: 50%;
+`;
+const StyledInsideCard = styled.div`
+  display: flex;
+`;
+const StyledPAdditives = styled.p`
+  display: flex;
+`;
+const StyledPAllergens = styled.p`
+  display: flex;
+`;
```

Comment:
this looks like repetitive code. How about having only one styled component that you use 3 times?

Reply:
for now you are right , but if i want to change things later on the layout and i will! I have to change that again to different parts . Also for me it is easier to read my code later .

Reply:
I understand what you mean. 
But wouldn't using variant-syntax with styledComponents serve the same purpose (it is even more clearly and can be expanded any time)?